src/main/java/me/jellysquid/mods/sodium/client/render/chunk/compile/ChunkBuilder.java

```diff
@@ -83,22 +112,42 @@ public void startWorkers() {
 
         MinecraftClient client = MinecraftClient.getInstance();
 
-        for (int i = 0; i < this.limitThreads; i++) {
+        for (int i = 0; i < this.initialThreads; i++) {
+            // Don't change any of this code (even though we factored it out into spawnThread()) because SeedQueue
+            // injects like crazy in here.
             ChunkBuildBuffers buffers = new ChunkBuildBuffers(this.vertexType, this.renderPassManager);
             ChunkRenderCacheLocal pipeline = new ChunkRenderCacheLocal(client, this.world);
 
             WorkerRunnable worker = new WorkerRunnable(buffers, pipeline);
 
-            Thread thread = new Thread(worker, "Chunk Render Task Executor #" + i);
+            Thread thread = new Thread(worker, "Chunk Render Task Executor #" + this.threads.size());
             thread.setPriority(Math.max(0, Thread.NORM_PRIORITY - 2));
             thread.start();
 
             this.threads.add(thread);
         }
 
+        // More SeedQueue compatibility, ideally we would log the number of threads.
         LOGGER.info("Started {} worker threads", this.threads.size());
     }
 
+    public void maybeIncreaseThreadLimit()
+    {
+        // Sometimes, we want more threads :)
+        // In that case, let's slowly add them!
+        // :) Doesn't everyone love more threads?
+        if (!this.hasThreadSpace || (this.updates++ < 60 /* Arbitrary! */) || !this.running.get()) {
```

Comment:
Completely arbitrary value. I'm not sure if calls of this function are tied to ticks or to render updates (i.e. fps). I think it's likely the latter. In that case, we almost want to just use a clock, but I did it this way because polling system clocks can be a bit less performant.